src/libraries/System.Diagnostics.DiagnosticSource/src/System/Diagnostics/Metrics/Meter.cs

```diff
@@ -14,23 +14,66 @@ public class Meter : IDisposable
     {
         private static readonly List<Meter> s_allMeters = new List<Meter>();
         private List<Instrument> _instruments = new List<Instrument>();
+        private Dictionary<string, List<Instrument>> _nonObservableInstrumentsCache = new();
         internal bool Disposed { get; private set; }
 
+        /// <summary>
+        /// Initialize a new instance of the Meter using the <see cref="MeterOptions" />.
+        /// </summary>
+        public Meter(MeterOptions options)
+        {
+            if (options is null)
+            {
+                throw new ArgumentNullException(nameof(options));
+            }
+
+            Debug.Assert(options.Name != null);
+
+            Initialize(options.Name, options.Version, options.Tags, options.Scope);
+
+            Debug.Assert(Name is not null);
+        }
+
         /// <summary>
         /// Initializes a new instance of the Meter using the meter name.
         /// </summary>
         /// <param name="name">The Meter name.</param>
-        public Meter(string name) : this (name, null) {}
+        public Meter(string name) : this (name, null, null, null) {}
+
+        /// <summary>
+        /// Initializes a new instance of the Meter using the meter name and version.
+        /// </summary>
+        /// <param name="name">The Meter name.</param>
+        /// <param name="version">The optional Meter version.</param>
+        public Meter(string name, string? version) : this (name, version, null, null) {}
 
         /// <summary>
         /// Initializes a new instance of the Meter using the meter name and version.
         /// </summary>
         /// <param name="name">The Meter name.</param>
         /// <param name="version">The optional Meter version.</param>
-        public Meter(string name, string? version)
+        /// <param name="tags">The optional Meter tags.</param>
+        /// <param name="scope">The optional Meter scope.</param>
+        /// <remarks>
+        /// You can use the scope object to link several Meters with a particular scope.
+        /// For instance, a dependency injection container can choose to associate all Meters that are created within the container with its own scope.
+        /// If the scope object is null, it indicates that the Meter is not linked to any particular scope.
+        /// </remarks>
+        public Meter(string name, string? version, IEnumerable<KeyValuePair<string, object?>>? tags, object? scope = null)
+        {
+            Initialize(name, version, tags, scope);
+            Debug.Assert(Name is not null);
+        }
+
+        private void Initialize(string name, string? version, IEnumerable<KeyValuePair<string, object?>>? tags, object? scope = null)
         {
             Name = name ?? throw new ArgumentNullException(nameof(name));
             Version = version;
+            if (tags is not null)
+            {
+                Tags = new List<KeyValuePair<string, object?>>(tags);
```

Comment:
The consumer could do:
```C#
((List<...>)meter.Tags).Add(...);
```
Would that mess anything up?

Reply:
Users doing that will be missing up with themselves. It is not good to depend on the internal implementation anyway. But even if they do that in this case, it will not cause crashes or terrible things. Will just be updating a list that not supposed to be mutated.